### PR TITLE
Update testing to a new approach with CloudBuild

### LINKED ISSUE #14: Configure CI pipeline
- [ ] define Terraform test fixture infrastructure in https://github.com/googlecloudplatform/cloud-foundation-toolkit
- [ ] define Concourse pipeline in https://github.com/googlecloudplatform/cloud-foundation-toolkit
- [ ] update `Makefile`, `test/ci_integration.sh`, and related files based on https://github.com/terraform-google-modules/terraform-google-module-template.

### LINKED ISSUE #16: Add sample to README
Please add sample code to use this module in the main README.md

### LINKED ISSUE #30: Update testing to new approach
We should update the testing to use Cloud Build and a new Service Account per run, as documented [here](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0).

### PR DESCRIPTION
**Pull request by averbuks**
Update testing to a new approach with CloudBuild (fixes #30 and fixes #14)
Fixed all the linting issues
Basic usage example added to README.md (fixes #16)

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,309 +0,0 @@
-GEM
```
Comment (morgante) [inline Gemfile.lock]: I don't think we're meant to delete this file?

File: README.md [hunk 2]
```diff
@@ -54,31 +94,43 @@ This module provisions a dataset and a table with an associated JSON schema.
...
-`xcode-select --install`
+
+These sections describe requirements for using this module.
+
+### Software
+
+The following dependencies must be available:
+
+- [Terraform][terraform] v0.12
+- [Terraform Provider for GCP][terraform-provider-gcp] plugin v2.0
```
Comment (morgante) [inline README.md]: We should provide the minor version number (v2.5).

File: README.md [hunk 3]
```diff
@@ -54,31 +94,43 @@ This module provisions a dataset and a table with an associated JSON schema.
...
-### Terraform plugins
-- [Terraform](https://www.terraform.io/downloads.html) 0.12.x
-- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) plugin v2.5.0
-
-### Permissions
-In order to execute this module you must have a Service Account with the following roles:
- - roles/bigquery.dataOwner
-
-#### Script Helper
-A helper script for configuring a Service Account is located at (./helpers/setup-sa.sh).
```
Comment (morgante) [inline README.md]: Let's also ckeep the reference to the helper script.

File: build/lint.cloudbuild.yaml [hunk 4]
```diff
@@ -11,3 +11,14 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+steps:
+- name: 'gcr.io/cloud-foundation-cicd/cft/developer-tools:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
```
Comment (morgante) [inline build/lint.cloudbuild.yaml]:
```suggestion
- name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
```

File: Gemfile.lock (hunk 1 above)
Comment (averbuks) [reply Gemfile.lock]: Looks like `Gemfile.lock` supposed be deleted according to the [instruction](https://docs.google.com/document/d/1eiHpZlhEDMDG8gi4LHyv2aZs1TJmeFlF8sqCE57_j4A/edit#heading=h.rz8l5qn0ygc0)

File: README.md (hunk 2 above)
Comment (averbuks) [reply README.md]: Updated

File: README.md (hunk 3 above)
Comment (averbuks) [reply README.md]: Yes, makes sense, updated

File: build/lint.cloudbuild.yaml (hunk 4 above)
Comment (averbuks) [reply build/lint.cloudbuild.yaml]: done

**Pull request merged**